feature/qos/otg_tests/ingress_police_nhg/README.md

```diff
@@ -0,0 +1,363 @@
+              bc: 100000                # 100 kilobytes
+              queuing-behavior: POLICE
+            exceed-action:
+              config:
+                drop: TRUE
+
+  interfaces:                  # this is repeated per subinterface (vlan)
+    - interface: "PortChannel1.100"
+        config:
+          interface-id: "PortChannel1.100"
+        input:
+          classifers:
+            - classifier:
+              config:
+                name: "dest_A"
+                type: "IPV4"
+            scheduler-policy:
+              state:
+                name: limit_group_A_1Gb
+    - interface: "PortChannel1.200"
+        config:
+          interface-id: "PortChannel1.200"
+        input:
+          classifers:
+            - classifier:
+              config:
+                name: "dest_B"
+                type: "IPV4"
+          scheduler-policy:
+            state:
+              name: limit_group_B_2Gb
+
+            scheduler-policy:
+              state:
+                name: limit_group_A_1Gb
+
+```
+
+### TE-18.2.2 push gRIBI aft encap rules with next-hop-group-id
+
+Create a gRIBI client and send this proto message to the DUT to create AFT
+entries.  Note the next-hop-groups here include a `next_hop_group_id` field
+which matches the
+`/qos/classifiers/classifier/condition/next-hop-group/config/name` leaf.
+
+* [TODO: OC AFT Encap PR in progress](https://github.com/openconfig/public/pull/1153)
+* [TODO: gRIBI v1 protobuf defintions](https://github.com/openconfig/gribi/blob/master/v1/proto/README.md)
+
+```proto
+network_instances: {
+  network_instance: {
+    afts {
+      #
+      # entries used for "group_A"
+      ipv6_unicast {
+        ipv6_entry {
+          prefix: "inner_ipv6_dst_A"   # this is an IPv6 entry for the origin/inner packet
+          next_hop_group: 100
+        }
+      }
+      ipv4_unicast {
+        ipv4_entry {
+          prefix: "ipv4_inner_dst_A"   # this is an IPv4 entry for the origin/inner packet
+          next_hop_group: 100
+        }
+      }
+      next_hop_groups {
+        next_hop_group {
+          id: 100
+          next_hop_group_id: "nhg_A"  # new OC path /network-instances/network-instance/afts/next-hop-groups/next-hop-group/state/next-hop-group-id
+          next_hops {                 # reference to a next-hop
+            next_hop: {
+              index: 100
+            }
+          }
+        }
+      }
+      next_hops {
+        next_hop {
+          index: 100
+          network_instance: "group_A"
+          encap-headers {
+            encap-header {
+              index: 1
+              pushed_mpls_label_stack: [100,]
+            }
+          }
+          encap-headers {
+            encap-header {
+              index: 2
+              src_ip: "outer_ipv6_src"
+              dst_ip: "outer_ipv6_dst_A"
+              dst_udp_port: "outer_dst_udp_port"
+              ip_ttl: "outer_ip-ttl"
+              dscp: "outer_dscp"
```

Comment:
Darren,, Not now, but just wanted to make sure this is the section that will be updated with IPsec SPI detail. correct ?

Reply:
IPSec is covered by a separate test at [TE-18.6](https://github.com/openconfig/featureprofiles/pull/3435)

We wanted to scope the tests separately so we can show 
TE-18.1: encap/decap works
TE-18.2: encap/decap + policer works
TE-18.6: encap/decap + policer +ipsec works

If we put all of these into one test, then it would be hard to tell if the different important parts are working without debugging.